yt/loaders.py

```diff
@@ -1144,7 +1159,7 @@ def load_octree(
         parameters=parameters,
     )
 
-    handler.name = "OctreeData"
+    handler.name = basename
```

Comment:
You're getting those type checking failures cause you added the `str` annotation to the new argument. I think fixing the type checking here is well beyond the scope of your PR (you're only exposing it here because this is the first type-annotated argument to this function), so I think it's fine to simply add more ignore directives similar to the other loaders. For this line, you'll want:

```suggestion
    handler.name = basename  # type: ignore [attr-defined]
```

But then you'll also need to add that `# type: ignore [attr-defined]` comment to all of the other lines mentioned in the test failure too. Here are the lines for reference (copied from [here](https://github.com/yt-project/yt/actions/runs/4331206959/jobs/7562899186)):

```
yt/loaders.py: note: In function "load_octree":
yt/loaders.py:1162: error: "StreamHandler" has no attribute "name"  [attr-defined]
yt/loaders.py:1163: error: "StreamHandler" has no attribute "domain_left_edge"  [attr-defined]
yt/loaders.py:1164: error: "StreamHandler" has no attribute "domain_right_edge"  [attr-defined]
yt/loaders.py:1165: error: "StreamHandler" has no attribute "refine_by"  [attr-defined]
yt/loaders.py:1166: error: "StreamHandler" has no attribute "dimensionality"  [attr-defined]
yt/loaders.py:1167: error: "StreamHandler" has no attribute "domain_dimensions"  [attr-defined]
yt/loaders.py:1168: error: "StreamHandler" has no attribute "simulation_time"  [attr-defined]
yt/loaders.py:1169: error: "StreamHandler" has no attribute "cosmology_simulation"  [attr-defined]
yt/loaders.py:1174: error: "StreamOctreeDataset" has no attribute "octree_mask"  [attr-defined]
yt/loaders.py:1175: error: "StreamOctreeDataset" has no attribute "partial_coverage"  [attr-defined]
yt/loaders.py:1176: error: "StreamOctreeDataset" has no attribute "num_zones"  [attr-defined]
```

Reply:
alternatively, you could change `basename: str = "OctreeData",` to `basename = "OctreeData"`, but I'm in slight favor of adding all the typing ignores. @neutrinoceros do you agree with that?

Reply:
I do !

Reply:
Thanks @chrishavlin ! I do see this error ignoring in the original code for all other types of datasets except `OctreeData`, so I was wondering whether there's a particular reason for that. But now as you suggested, I add those lines for `OctreeData` too.